pkg/data/data_test.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package data
+
+import (
+	"testing"
+)
+
+const (
+	dbFile = "testdata/boltdb/db"
+)
+
+func TestListKeySummariesFilters(t *testing.T) {
+	cases := []struct {
+		name         string
+		filters      []Filter
+		expectedKeys []string
+	}{
+		{
+			name:         "nofilters",
+			filters:      []Filter{},
+			expectedKeys: []string{"/registry/jobs/default/pi", "/registry/namespaces/default", "/registry/pods/default/pi-dqtsw", "compact_rev_key"},
+		},
+		{
+			name:         "prefixfilter",
+			filters:      []Filter{NewPrefixFilter("/registry/jobs")},
+			expectedKeys: []string{"/registry/jobs/default/pi"},
+		},
+		{
+			name:         "namespacefilter",
+			filters:      []Filter{mustBuildFilter(&FieldConstraint{lhs: ".Value.metadata.namespace", op: Equals, rhs: "default"})},
+			expectedKeys: []string{"/registry/jobs/default/pi", "/registry/pods/default/pi-dqtsw"},
+		},
```

Comment:
From the logic of the code, there could be both prefix filter and some other filter coexisting, would it worth to add a test with both?

Reply:
Yes, this is needed. Added.